### PR TITLE
field move base function

### PR DESCRIPTION
**Pull request by ghoulslash**
rewrite field move HM function from pokemenu BRM

### CODE REVIEW

File: src/fieldMoves/ValidateAndLaunchFieldmove.c [hunk 1]
```diff
@@ -0,0 +1,122 @@
...
+	
+	bool usable = gFieldMoves[fieldIndex].BRMPrepareFunc();
+	if (usable == false) {
+		u8 priority = 1;
+		BRMDisplayMessageTask(gFieldMoves[fieldIndex].errorMsg, priority);
+		tasks[task_id].function = TaskResetBRM;
+		return;	
+	}
+	else {
+		gFieldMoves[fieldIndex].LaunchFieldTask(fieldIndex, task_id);	//field moves, not flash/surf
```
Comment (EternalCode) [inline src/fieldMoves/ValidateAndLaunchFieldmove.c]: task_id to taskId

File: src/fieldMoves/ValidateAndLaunchFieldmove.c [hunk 2]
```diff
@@ -0,0 +1,122 @@
...
+};
+
+
+// func called from selecting HM in BRM
+void ValidateAndLaunchFieldmove(u8 task_id) {		
+	// get relevant brm data
+	u8 choice_ind = choice_curr;
+	//u8 choice_ind = choices.curr;	// TO DO
+	
+	u8* brm0 = (u8*) brm_data;
```
Comment (EternalCode) [inline src/fieldMoves/ValidateAndLaunchFieldmove.c]: camelcase var

File: src/fieldMoves/ValidateAndLaunchFieldmove.c [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	{BRMPrepareSoftboiled_MilkDrink, BRMLaunchSoftboiled_MilkDrink, MilkDrinkErrorMsg},
+	{BRMPrepareSoftboiled_MilkDrink, BRMLaunchSoftboiled_MilkDrink, MilkDrinkErrorMsg},
+	{BRMPrepareSweetScent, BRMLaunchExitOverworld, SweetScentErrorMsg},
+};
+
+
+// func called from selecting HM in BRM
+void ValidateAndLaunchFieldmove(u8 task_id) {		
+	// get relevant brm data
+	u8 choice_ind = choice_curr;
```
Comment (EternalCode) [inline src/fieldMoves/ValidateAndLaunchFieldmove.c]: camelcase var

File: src/fieldMoves/ValidateAndLaunchFieldmove.c [hunk 4]
```diff
@@ -0,0 +1,122 @@
...
+}
+
+void BRMLaunchSoftboiled_MilkDrink(u8 fieldIndex, u8 task_id) {
+	TaskSelectBRMPokeForHeal(task_id);
+	return;
+}
+
+void BRMLaunchExitOverworld(u8 fieldIndex, u8 task_id) {
+	brm_addr = exit_to_overworld_2_switch;
+	u8 brm_slot = brm_get_pokemon_selection();
```
Comment (EternalCode) [inline src/fieldMoves/ValidateAndLaunchFieldmove.c]: camelcase var

File: src/fieldMoves/ValidateAndLaunchFieldmove.c [hunk 5]
```diff
@@ -0,0 +1,122 @@
...
+	return;
+}
+
+void BRMLaunchSoftboiled_MilkDrink(u8 fieldIndex, u8 task_id) {
+	TaskSelectBRMPokeForHeal(task_id);
+	return;
+}
+
+void BRMLaunchExitOverworld(u8 fieldIndex, u8 task_id) {
+	brm_addr = exit_to_overworld_2_switch;
```
Comment (EternalCode) [inline src/fieldMoves/ValidateAndLaunchFieldmove.c]: camelcase var

File: src/fieldMoves/ValidateAndLaunchFieldmove.c [hunk 6]
```diff
@@ -0,0 +1,122 @@
...
+void BRMLaunchFly(u8 fieldIndex, u8 task_id) {
+	brm_addr = PerformBRMFly;	// store task in brm
+	TaskBRMPalFade(task_id);	
+	return;
+}
+
+void BRMLaunchTeleport(u8 fieldIndex, u8 task_id) {
+	u8 teleport_map = gSaveBlock1Ptr->lastHealLocation.map;
+	u8 teleport_bank = gSaveBlock1Ptr->lastHealLocation.bank;
+	AskAndDoBRMWarp( teleport_bank, teleport_map, task_id, fieldIndex, AskTeleport );
```
Comment (EternalCode) [inline src/fieldMoves/ValidateAndLaunchFieldmove.c]: camelcase vars

File: src/fieldMoves/ValidateAndLaunchFieldmove.c [hunk 7]
```diff
@@ -0,0 +1,122 @@
...
+	return;
+}
+
+void BRMLaunchDig(u8 fieldIndex, u8 task_id) {
+	u8 dig_map = gSaveBlock1Ptr->warp4.map;
+	u8 dig_bank = gSaveBlock1Ptr->warp4.bank;
+	AskAndDoBRMWarp( dig_bank, dig_map, task_id, fieldIndex, AskDig );
```
Comment (EternalCode) [inline src/fieldMoves/ValidateAndLaunchFieldmove.c]: camelcase var

File: src/fieldMoves/ValidateAndLaunchFieldmove.c [hunk 8]
```diff
@@ -0,0 +1,122 @@
...
+	return;
+}
+
+void BRMLaunchSoftboiled_MilkDrink(u8 fieldIndex, u8 task_id) {
```
Comment (EternalCode) [inline src/fieldMoves/ValidateAndLaunchFieldmove.c]: camelcase vars and parameters

**Comment by ghoulslash** [comment]
updated the vars in pokeagb and here to camelcase.

**Pull request closed without merging**